15-knitr-markdown.Rmd

```diff
@@ -0,0 +1,383 @@
+```
+
+
+
+
+### Chunk options
+
+There are a variety of options to affect how the code chunks are
+treated.
+
+- Use `echo=FALSE` to avoid having the code itself shown.
+- Use `results="hide"` to avoid having any results printed.
+- Use `eval=FALSE` to have the code shown but not evaluated.
+- Use `warning=FALSE` and `message=FALSE` to hide any warnings or
+  messages produced.
+- Use `fig.height` and `fig.width` to control the size of the figures
+  produced (in inches).
+
+So you might write:
+
+<pre>
+&#96;&#96;&#96;{r load_libraries, echo=FALSE, message=FALSE}
+library(dplyr)
+library(ggplot2)
+&#96;&#96;&#96;
+</pre>
+
+Often there'll be particular options that you'll want to use
+repeatedly; for this, you can set _global_ chunk options, like so:
+
+<pre>
+&#96;&#96;&#96;{r global_options, echo=FALSE}
+knitr::opts_chunk$set(fig.path="Figs/", message=FALSE, warning=FALSE,
+                      echo=FALSE, results="hide", fig.width=11)
+&#96;&#96;&#96;
+</pre>
+
+The `fig.path` option defines where the figures will be saved. The `/`
+here is really important; without it, the figures would be saved in
+the standard place but just with names that being with `Figs`.
+
+If you have multiple R Markdown files in a common directory, you might
+want to use `fig.path` to define separate prefixes for the figure file
+names, like `fig.path="Figs/cleaning-"` and `fig.path="Figs/analysis-"`.
+
+
+> ### Challenge {.challenge}
+>
+> Use chunk options to control the size of a figure and to hide the
+> code.
+
+
+### Inline R code
+
+You can make _every_ number in your report reproducible. Use
+<code>&#96;r</code> and <code>&#96;</code> for an in-line code chunk,
+like so: <code>&#96;r round(some_value, 2)&#96;</code>. The code will be
+executed and replaced with the _value_ of the result.
+
+Don't let these in-line chunks get split across lines.
+
+Perhaps precede the paragraph with a larger code chunk that does
+calculations and defines things, with `include=FALSE` for that larger
+chunk (which is the same as `echo=FALSE` and `results="hide"`).
+
+I'm very particular about rounding in such situations. I may want
+`2.0`, but `round(2.03, 1)` will give just `2`.
+
+The
+[`myround`](https://github.com/kbroman/broman/blob/master/R/myround.R)
+function in my [R/broman](https://github.com/kbroman) package handles
+this.
+
+> ### Challenge {.challenge}
+>
+> Try out a bit of in-line R code.
+
+
+### Other output options
+
+You can also convert R Markdown to a PDF or a Word document. Click the
+little triangle next to the "Knit HTML" button to get a drop-down
+menu. Or you could put `pdf_document` or `word_document` in the header
+of the file.
+
+> ## Tip: Creating PDF documents {.callout}
+>
+> Creating .pdf documents may require installation of some extra software. If required this is detailed in an error message.
+>
+> Tex for windows is available [here](http://miktex.org/2.9/setup).
+>
+> Tex for mac is available [here](http://tug.org/mactex).
```

Comment:
@tomwright01 Do you think it is worth discussing `spin()`, and the use of special comments that end up formatted as markdown (i.e. `#' a *markdown* comment`)? Or is that overload?